eulxml/__init__.py

```diff
@@ -14,9 +15,34 @@
 #   See the License for the specific language governing permissions and
 #   limitations under the License.
 
+import os
+import pkg_resources
+
 __version_info__ = (1, 1, 0, 'dev')
 
 # Dot-connect all but the last. Last is dash-connected if not None.
 __version__ = '.'.join([str(i) for i in __version_info__[:-1]])
 if __version_info__[-1] is not None:
     __version__ += ('-%s' % (__version_info__[-1],))
+
+# paths to xml schema catalog file & directory
+
+#: relative path for schema data directory
+SCHEMA_DATA_DIR = 'schema_data'
+
+# use package resources if possible, so this will work from an egg
+# http://peak.telecommunity.com/DevCenter/PythonEggs#accessing-package-resources
+if pkg_resources.resource_isdir(__name__, SCHEMA_DATA_DIR):
+    XMLCATALOG_DIR = pkg_resources.resource_filename(__name__,
+                                                     SCHEMA_DATA_DIR)
+    XMLCATALOG_FILE = pkg_resources.\
+        resource_filename(__name__,
+                          '%s/catalog.xml' % SCHEMA_DATA_DIR)
+else:
+    XMLCATALOG_DIR = os.path.join(os.path.dirname(__file__),
+                                  SCHEMA_DATA_DIR)
+    XMLCATALOG_FILE = os.path.join(XMLCATALOG_DIR, 'catalog.xml')
+
+os.environ['XML_CATALOG_FILES'] = ":".join((filter(None, (os.environ.get('XML_CATALOG_FILES'), XMLCATALOG_FILE))))
```

Comment:
I know I wrote a lot of this, but - might be worth adding a more comments here so it's very clear why this code is here and what it does, so we don't have to try to remember or look it up the next time we need to touch it.